pkg/sql/sem/builtins/builtins.go

```diff
@@ -336,6 +336,27 @@ var builtins = map[string]builtinDefinition{
 				return tree.NewDInt(tree.DInt(bit)), nil
 			},
 			Info: "Extracts a bit at given index in the bit array",
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"byte_string", types.Bytes}, {"index", types.Int}},
+			ReturnType: tree.FixedReturnType(types.Int),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				byteString := []byte(*args[0].(*tree.DBytes))
+				index := int(tree.MustBeDInt(args[1]))
+				// Check whether index asked is inside ByteArray.
+				if index < 0 || index >= 8*len(byteString) {
+					return nil, pgerror.Newf(pgcode.ArraySubscript,
+						"bit index %d out of valid range (0..%d)", index, 8*len(byteString)-1)
+				}
+				// To extract a bit at the given index, we have to determine the
+				// position within byte array, i.e. index/8 after that checked
+				// the bit at residual index.
+				if byteString[index/8]&(byte(1)<<(8-1-byte(index)%8)) != 0 {
```

Comment:
We could replace `8` with a `globally constant variable` or by `len(byte)` but avoided because, it very less likely it would be modified in future, and we had previously used it as `8` instead of defining as some constant.
```
if byteString[index/8]&(byte(1)<<(8-1-byte(index)%8)) != 0 {
```
use `8-1` instead of `7` thought it would be more understandable.

Reply:
i think this is fine :)